packages/flutter_tools/lib/src/application_package.dart

```diff
@@ -48,6 +50,35 @@ class AndroidApk extends ApplicationPackage {
     assert(launchActivity != null);
   }
 
+  /// Creates a new AndroidApk from an existing APK.
+  factory AndroidApk.fromApk(String applicationBinary) {
+    String aaptPath = androidSdk?.latestVersion?.aaptPath;
+    if (aaptPath == null) {
+      printError('Unable to locate the Android SDK; please run \'flutter doctor\'.');
+      return null;
+    }
+
+    List<String> aaptArgs = <String>[aaptPath, 'dump', 'badging', applicationBinary];
+    ApkManifestData data = ApkManifestData.parseFromAaptBadging(runCheckedSync(aaptArgs));
+
+    if (data == null) {
+      printError('Unable to read manifest info from $applicationBinary.');
+      return null;
+    }
+
+    if (data.packageName == null || data.launchableActivityName == null) {
+      printError('Unable to read manifest info from $applicationBinary.');
+      return null;
+    }
+
+    return new AndroidApk(
+      buildDir: 'build',
```

Comment:
It looks like this is only used for ios `ApplicationPackage`s; we could probably refactor to remove it from the common parent class (of the android and ios subclasses).